pennylane/measurements/sample.py

```diff
@@ -198,7 +198,10 @@ def shape(self, device, shots):
                 "Shots are required to obtain the shape of the measurement "
                 f"{self.__class__.__name__}."
             )
-        len_wires = len(self.wires) if len(self.wires) > 0 else len(device.wires)
+        if self.obs:
+            len_wires = 1
```

Comment:
This is just needed to make the dtype shape struct right? I'm assuming that this works because most of the time multi-wire observables are tensors of Paulis/Hadamard with only two non-degenerate eigenvalues, but could it cause problems with multi wire observables that have *more* than two non-degenerate eigenvalues? If that works, great! But a test case for that would be a good idea.

Reply:
This method is to calculate the shape of the resulting ndarray.  When measuring an observable, we are sampling from the eigenvalues of that observable, so we only get one number per sample.  

Maybe we could rename the `len_wires` variable to something like `values_per_shot` or something like that?